File: Java/Tests/jdi-uitest-wintests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/pages/ContactFormTwoButtons.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.pages;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.testing.unittests.Contact;
import com.epam.jdi.uitests.win.winium.elements.common.TextArea;
import com.epam.jdi.uitests.win.winium.elements.common.TextField;
import com.epam.jdi.uitests.win.winium.elements.composite.Form;
import org.openqa.selenium.support.FindBy;

import java.util.Arrays;
import java.util.List;

/**
 * Created by Dmitry_Lebedev1 on 10/16/2015.
 */
public class ContactFormTwoButtons extends Form<Contact> implements IContactForm {

    @FindBy(id = "Name")
    public TextField name;

    @FindBy(id = "LastName")
    public TextField lastName;

    @FindBy(id = "Description")
    public TextArea description;

    @FindBy(xpath = "//*[text()='Submit']")
    public IButton submit;

    @FindBy(xpath = "//*[text()='Calculate']")
    public IButton calculate;

    public List<String> getFormValue() {
        return Arrays.asList(
                name.getWebElement().getAttribute("value"),
                lastName.getWebElement().getAttribute("value"),
                description.getWebElement().getAttribute("value"));
    }

    public void fillForm(Contact contact) {
        name.getWebElement().clear();
        lastName.getWebElement().clear();
        description.getWebElement().clear();
        name.getWebElement().sendKeys(contact.name);
        lastName.getWebElement().sendKeys(contact.lastName);
        description.getWebElement().sendKeys(contact.description);
    }
}

File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/UIObjects/Sections/ContactFormTwoButtons.cs
﻿using Epam.JDI.Core.Interfaces.Common;
using JDI_UIWebTests.Entities;
using JDI_Web.Attributes;
using JDI_Web.Selenium.Elements.Common;
using JDI_Web.Selenium.Elements.Composite;
using System.Collections.Generic;


namespace JDI_UIWebTests.UIObjects.Sections
{
    public class ContactFormTwoButtons : Form<Contact>
    {
        [FindBy(Id = "Name")]
        public new TextField Name;

        [FindBy(Id = "LastName")]
        public TextField LastName;

        [FindBy(Id = "Description")]
        public TextArea Description;

        [FindBy(XPath = "//*[text()='Submit']")]
        public new IButton Submit;

        [FindBy(XPath = ".//a[@class='ui-slider-handle ui-state-default ui-corner-all' and position()=1]")]
        public Link FirstRoller;

        [FindBy(XPath = ".//a[@class='ui-slider-handle ui-state-default ui-corner-all' and position()=2]")]
        public Link SecondRoller;

        public List<string> GetFormValue()
        {
            var fields = new List<string>
            {
                Name.GetText,
                LastName.GetText,
                Description.GetText
            };
            return fields;
        }

        public void FillForm(Contact contact)
        {
            Name.NewInput(contact.FirstName);
            LastName.NewInput(contact.LastName);
            Description.NewInput(contact.Description);
        }
    }
}
